src/server/src/main/java/io/cassandrareaper/service/RepairRunner.java

```diff
@@ -407,12 +410,15 @@ private void startNextSegment() throws ReaperException, InterruptedException {
                 repairRunId);
 
     Optional<RepairSegment> nextRepairSegment = Optional.empty();
+    Collection<String> potentialReplicas = new HashSet<>();
     for (RepairSegment segment : nextRepairSegments) {
-      Collection<String> potentialReplicas = repairUnit.getIncrementalRepair()
+      Map<String, String> potentialReplicaMap = this.repairRunService.getDCsByNodeForRepairSegment(
+          cluster, segment.getTokenRange(), repairUnit.getKeyspaceName(), repairUnit);
+      potentialReplicas = repairUnit.getIncrementalRepair()
           ? Collections.singletonList(segment.getCoordinatorHost())
```

Comment:
That's actually a very good question :)
To give a little bit of context, incremental repairs create "fake" segments in Reaper. One per host, because it has to run on all ranges of each node, not on subranges.
Hence, when the repair run is created, the coordinator host for each segment is set and it cannot be recomputed afterwards like we do for token ranges.
We yet have to get issues raised on that particular issue, but I'd consider it outside of the scope for this PR, since the implementation of incremental repair is fairly different.
I guess a workaround could be to use host ids instead of ip addresses for the coordinator hosts and resolve the IP address right before triggering the segments.